Source/cs505/group1/state/ButtonContext.java

```diff
@@ -0,0 +1,62 @@
+package cs505.group1.state;
+
+/**
+ * An abstract framework for developing and using contexts in a finite state machine
+ * to set actions generated by a GrovePi button sensor.
+ * @author CS505 Fall17 Group1: <br>
+ * Emily Park<br>
+ * Jeffrey Blankenship  <br>
+ * Cecelia Oluwadoyinsola<br>
+ * James Luczynski<br>
+ * Melissa Mulcahy <br>
+ * @version 2017.10.29
+ */
+public abstract class ButtonContext {
+ 
+    //No javadoc provided: private visibility.
+    //This instance variable stores the current buttonState.
+    private ButtonState buttonState;
+    
+    /**
+     * Constructs a newly allocated ButtonContext object that represents the specified buttonState.
+     * 
+     * @param buttonState sets the initial value of buttonState.
+     */
+    protected ButtonContext(ButtonState buttonState){
+        this.buttonState = buttonState;
+    }
+    
+    /**
+     * Implements code for a single button press.
+     */
+    public abstract void singlePress();
+    
+    /**
+     * Implements code for a double button press.
+     */
+    public abstract void doublePress();

```

Comment:
Rather than make all of these abstract since the base functionality of the State pattern is the context passing the call onto the current state make that the default behavior would make sense.